lib/config-generator.js

```diff
@@ -0,0 +1,260 @@
+
+                exportContent += `  ...compat.extends("${extendedName}"),\n`;
+            }
+        }
+
+        if (this.answers.module === "commonjs" || this.answers.module === "script") {
+            languageOptionsContent += `sourceType: "${this.answers.module}", `;
+        }
+
+        if (this.answers.env?.length > 0) {
+            this.result.devDependencies.push("globals");
+            importContent += "import globals from \"globals\";\n";
+            const envContent = {
+                browser: "globals: globals.browser",
+                node: "globals: globals.node",
+                "browser,node": "globals: {...globals.browser, ...globals.node}"
+            };
+
+            languageOptionsContent += `${envContent[this.answers.env.join(",")]}, `;
+        }
+
+
+        if (this.answers.language === "typescript") {
+            this.result.devDependencies.push("typescript-eslint");
+            importContent += "import tseslint from \"typescript-eslint\";\n";
+            exportContent += "  ...tseslint.configs.recommended,\n";
+        }
+
+        if (this.answers.framework === "vue") {
+
+            this.result.devDependencies.push("eslint-plugin-vue");
+
+            // importContent += "import pluginVue from \"eslint-plugin-vue\";\n";
+
+            // // TODO: there is a wip for flat support - https://github.com/vuejs/eslint-plugin-vue/pull/2319
+            // exportContent += "  pluginVue.configs[\"flat/essential\"],\n";
+            needCompatHelper = true;
+            exportContent += "  ...compat.extends(\"plugin:vue/vue3-essential\").map(config => ({files: [\"**/*.vue\"], ...config})),\n";
+        }
+
+        if (this.answers.framework === "react") {
+            this.result.devDependencies.push("eslint-plugin-react");
+            importContent += "import pluginReactConfig from \"eslint-plugin-react/configs/recommended.js\";\n";
+            exportContent += "  pluginReactConfig,\n";
+        }
+
+        if (needCompatHelper) {
+            this.result.devDependencies.push("@eslint/eslintrc", "@eslint/js");
+        }
+        this.result.configContent = `${importContent}\n${needCompatHelper ? helperContent : ""}
+export default [\n  {languageOptions:{${languageOptionsContent}}},\n${exportContent}];`;
+    }
+
+    /**
+     * Output the configuration.
+     * @returns {void}
+     */
+    async output() {
+        const packageManager = (await enquirer.prompt({
+            type: "select",
+            name: "packageManager",
+            message: "Which package manager do you want to use?",
+            initial: 0,
+            choices: ["npm", "yarn", "pnpm", "bun"]
+        })).packageManager;
+
+        installSyncSaveDev(this.result.devDependencies, packageManager);
+
+        const configPath = path.join(this.cwd, this.result.configFilename);
+
+        await writeFile(configPath, this.result.configContent);
```

Comment:
When `package.json` has `"type": "module"`, this will (re-)create a `eslint.config.mjs` file. But if the user already has a `eslint.config.js` file, the .js file will take precedence, and the new .mjs file will be ignored.

Maybe we should just delete `eslint.config.js` in this case, since this tool does not preserve existing configuration anyway?

Reply:
it seems a very rare case - the package was designed for projects with no eslint configuration yet.🤔